src/lib/rsa/sandboxed/RSAKeysIframe.js

```diff
@@ -0,0 +1,85 @@
+/* global forge */
+
+/**
+ * @typedef {{command: 'generateKey', seed: string, keySize: number}} GenerateKeyCommand
+ * ...add new command types here and add them to the `Command` union type below
+ *
+ * @typedef {GenerateKeyCommand} Command
+ */
+
+const $loading = /** @type {HTMLSpanElement} */ (document.querySelector('#loading'));
+
+/**
+ * @param {any} key
+ * @returns {ArrayBuffer}
+ */
+function key2ArrayBuffer(key) {
+    const bytes = /** @type {string} */ (forge.asn1.toDer(key).getBytes());
+    const buffer = new ArrayBuffer(bytes.length);
+    const view = new Uint8Array(buffer);
+    for (let i = 0; i < view.length; i++) {
+        view[i] = bytes.charCodeAt(i);
+    }
+    return buffer;
+}
+
+window.addEventListener('message', async event => {
+    if (event.source === event.target) {
+        // console.log("Ignored same-window event:", event);
+        return;
+    }
+
+    /** @type {Command} */
+    const data = event.data;
+    if (!('command' in data) || data.command !== 'generateKey') return;
+    console.debug('RSA iframe MSG:', data);
+
+    $loading.textContent = '(generating...)';
+    await new Promise(res => window.setTimeout(res, 50)); // Give DOM time to update
+
+    // PRNG-seeded RSA key generation
+    const prng = forge.random.createInstance();
+    prng.seedFileSync = needed => {
+        // console.log('Need', needed, 'bytes of randomness, have', data.seed.length);
+        let seed = '';
+        while (seed.length < needed) {
+            seed += data.seed;
+        }
```

Comment:
If not enough entropy was provided, we should throw and reject instead. The caller should make sure to provide enough entropy or stretch it out with a proper key derivation function, instead of us simply repeating the seed here.
```suggestion
        if (needed > seed.length) {
            throw new Error(`Seed does not provide enough entropy. ${needed} bytes needed`);
        }
```

Reply:
Resolved in #524